### PR TITLE
[WIP] Error handling

### PR DESCRIPTION
**Pull request by lightsprint09**

### CODE REVIEW

File: Station/Station.js [hunk 1]
```diff
@@ -22,40 +29,58 @@ class Station {
...
+
     if (station.DBinformation) {
       this.DBInformationOpeningTimes = new OpeningTimes(station.DBinformation.availability);
     }
+
     if (station.localServiceStaff) {
       this.localServiceStaffAvailability = new OpeningTimes(station.localServiceStaff.availability);
     }
+
     this.primaryEvaId = station.evaNumbers.filter(eva => eva.isMain)[0].number;
```
Comment (dennispost) [inline Station/Station.js]: It is possible, that a station does not have any evaNumbers. Therefore you have to check for existence of `[0]`. Eg. `this.primaryEvaId = access(station.evaNumbers.filter(eva => eva.isMain)[0],'number');`

File: Station/Station.js [hunk 2]
```diff
@@ -22,40 +29,58 @@ class Station {
...
-    this.primaryRil100 = station.ril100Identifiers.filter(ril => ril.isMain)[0].rilIdentifier;
-    this.hasSteamPermission = station.ril100Identifiers.filter(ril => ril.isMain)[0].hasSteamPermission
-    this.priceCategory = station.priceCategory
-    this.hasWiFi = station.hasWiFi
-    this.hasTravelCenter = station.hasTravelCenter
-    this.hasRailwayMission = station.hasRailwayMission
-    this.hasDBLounge = station.hasDBLounge
-    this.hasLostAndFound = station.hasLostAndFound
-    this.hasCarRental = station.hasCarRental
+    let ril100Identifiers = (station.ril100Identifiers || []).filter(ril => ril.isMain)[0]
```
Comment (dennispost) [inline Station/Station.js]: Same as evaNumbers: check for existence of [0].

File: Station/Station.js [hunk 3]
```diff
@@ -22,40 +29,58 @@ class Station {
...
+    this.primaryRil100 = (ril100Identifiers || {}).rilIdentifier;
+    this.hasSteamPermission = (ril100Identifiers || {}).hasSteamPermission;
```
Comment (dennispost) [inline Station/Station.js]: hasSteamPermission is non-nullable

**Review by dennispost — CHANGES_REQUESTED** [review]

File: Station/Station.js (hunk 2 above)
Comment (lightsprint09) [reply Station/Station.js]: I think there is always an evaId

**Pull request merged**